sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -523,6 +524,14 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
     sortTest()
   }
 
+  test("negative in LIMIT or TABLESAMPLE") {
```

Comment:
Since we already have a large pr, if this test case is not related to the pr, we can submit another one for it.